packages/aggregator/src/analytics/aggregateAnalytics/adjustOptionsToAggregationList.js

```diff
@@ -93,12 +95,15 @@ const getAdjustedOrganisationUnitsAndAggregations = async (
       );
 
       if (!adjustedOrganisationUnitCodes) {
-        adjustedOrganisationUnitCodes = dataSourceEntities;
+        adjustedOrganisationUnitCodes =
+          dataSourceEntities.length > 0 ? dataSourceEntities : organisationUnitCodes;
+      }
+      if (Object.keys(relations).length > 0) {
+        adjustedAggregations.push({
+          ...aggregation,
+          config: { ...aggregation.config, orgUnitMap: relations },
+        });
```

Comment:
I think this is a safe assumption to make. If there are no relations, we don't need to include this aggregation as each analytic would just map to itself.